core/trino-main/src/main/java/io/trino/sql/routine/SqlRoutineAnalyzer.java

```diff
@@ -223,40 +223,40 @@ private static void validateArguments(FunctionSpecification function)
         }
     }
 
-    private static Optional<String> getLanguage(FunctionSpecification function)
+    private static Optional<Identifier> getLanguage(FunctionSpecification function)
     {
         List<LanguageCharacteristic> language = function.getRoutineCharacteristics().stream()
                 .filter(LanguageCharacteristic.class::isInstance)
                 .map(LanguageCharacteristic.class::cast)
-                .collect(toImmutableList());
+                .toList();
 
         if (language.size() > 1) {
-            throw semanticException(SYNTAX_ERROR, function, "Multiple language clauses specified");
+            throw semanticException(SYNTAX_ERROR, language.getLast(), "Multiple language clauses specified");
         }
 
         return language.stream()
                 .map(LanguageCharacteristic::getLanguage)
-                .map(Identifier::getValue)
                 .findAny();
     }
 
     private static void validateLanguage(FunctionSpecification function)
     {
-        Optional<String> language = getLanguage(function);
-        if (language.isPresent() && !language.get().equalsIgnoreCase("sql")) {
-            throw semanticException(NOT_SUPPORTED, function, "Unsupported language: %s", language.get());
-        }
+        getLanguage(function).ifPresent(language -> {
+            if (!language.getValue().equalsIgnoreCase("sql")) {
+                throw semanticException(NOT_SUPPORTED, language, "Unsupported function language: %s", language.getCanonicalValue());
+            }
+        });
```

Comment:
Is this refactor or part of the fix? ("Fix position reporting for routine characteristics")

Reply:
This is changed because `getLanguage()` now returns `Optional<Identifier>` so that the position can be reported